addons/calendar/calendar.py

```diff
@@ -1449,14 +1459,24 @@ def get_interval(self, cr, uid, ids, date, interval, tz=None, context=None):
             date = date.replace(tzinfo=pytz.timezone('UTC')).astimezone(timezone)
 
         if interval == 'day':
-            res = str(date.day)
+            # Day number (1-31)
+            res = unicode(date.day)
+
         elif interval == 'month':
-            res = date.strftime('%B') + " " + str(date.year)
+            # Localized month name and year
+            res = babel.dates.format_date(date=date, format='MMMM Y', locale=context.get('lang', 'en_US'))
+
         elif interval == 'dayname':
-            res = date.strftime('%A')
+            # Localized day name
+            res = babel.dates.format_date(date=date, format='EEEE', locale=context.get('lang', 'en_US'))
+
         elif interval == 'time':
+            # time
+
             dummy, format_time = self.get_date_formats(cr, uid, context=context)
-            res = date.strftime(format_time + " %Z")
+            res = unicode(date.strftime(format_time + " %Z"))
```

Comment:
Isn't there a chance that `format_time` includes unicode characters? IIRC Chinese locales use a 12-hour am/pm scheme, but the am/pm markers are unicode (上午 and 下午). I think it would be much safer to use `tools.ustr()` here.